### DISCUSSION TITLE
Virtual scrolling for TreeTable

### DISCUSSION THREAD

**Discussion by pablo-redradix** [primefaces/community#1874] [PrimeReact] [answered]
As of today, [TreeTable in PrimeReact](https://primereact.org/treetable/) lacks scroll virtualization options. Oddly enough, [its Angular counterpart does](https://primereact.org/treetable/)...

Are there any plans to implement this in the (near) future? I'm currently in a project where performance with large, hierarchical data sets is a top priority. We'll need to consider switching to a different library unless this is in the roadmap.

Thanks!

**Comment by melloware** [comment]
See: https://github.com/primefaces/primereact/issues/1423

**Reply by pablo-redradix** [reply]
Actually I did before I posted. Sadly, it doesn't provide much information: it took 3+ years for someone in the team to respond with a generic “this is not a defect”, hence my question.

**Reply by melloware** [reply]
you would probably have to financially sponsor this feature via PRO support: https://primereact.org/support/

**Comment by nitrogenous** [comment, marked as answer]
Thank you for your interest in the PrimeReact library. We appreciate your innovative suggestion. At present, our team is focused on upgrading the library to its next-generation version. Consequently, we have temporarily paused the addition of new features and components. However, please be assured that your idea has been noted and will be considered for inclusion in the forthcoming next-generation PrimeReact library.
